File: Java/src/main/java/com/gildedrose/LegendaryItem.java
package com.gildedrose;

/**
 *
 * @author markd
 */
public class LegendaryItem extends GenericItem{
    
    private final int DEFAULT_QUALITY = 80;
    
    public LegendaryItem(String name, int sellIn, int quality) {
        super(name, 0, quality);
        QUALITY_CHANGE_BEFORE_SELLBY = 0;
        QUALITY_CHANGE_AFTER_SELLBY = 0;
    }
    
    public LegendaryItem(String name) {
        super(name, 0, 0);
        quality = DEFAULT_QUALITY;
        qualityRating.set(DEFAULT_QUALITY);
        QUALITY_CHANGE_BEFORE_SELLBY = 0;
        QUALITY_CHANGE_AFTER_SELLBY = 0;
    }
    
    @Override
    public void updateQuality(){
        sellIn++;
        super.updateQuality();
    }
}

File: cpp-catch2/src/LegendaryItem.cpp
//
// Created by markd on 3/9/2021.
//

#include "LegendaryItem.h"

LegendaryItem::LegendaryItem(string name, int sellIn, int quality) : GenericItem(name, sellIn, quality) {
    sellIn = 0;
    QUALITY_CHANGE_BEFORE_SELLBY = 0;
    QUALITY_CHANGE_AFTER_SELLBY = 0;
}

void LegendaryItem::updateQuality() {
    sellIn++;
    GenericItem::updateQuality();
}
